src/core/components/start.js

```diff
@@ -3,19 +3,20 @@
 const series = require('async/series')
 const Bitswap = require('ipfs-bitswap')
 const FloodSub = require('libp2p-floodsub')
+const setImmediate = require('async/setImmediate')
 
 module.exports = (self) => {
   return (callback) => {
     callback = callback || function noop () {}
 
     const done = (err) => {
       if (err) {
-        self.emit('error', err)
+        setImmediate(() => self.emit('error', err))
```

Comment:
Note that callbacks are already async, it just happens that when emitting and event, 'everything else' would happen before the callback if there was no setImmediate